R/read_servers_helpers.R

```diff
@@ -102,56 +94,43 @@ identify_table_name <- function(query, tables) {
 #' @param password the user's password
 #' @param port the server port ID
 #'
-#' @return a list with the data fetched from the tables of interest
-#' @export
+#' @return a `list` of 1 or more objects of type `data.frame` containing each
+#'    data fetched from one table.
 #'
 #' @examples
+#' \dontrun{
 #' result <- fetch_data_from_query(
-#'  source = "select author_id, name, last_name from author",
-#'  dbms = "MySQL",
-#'  tables = c("family_author", "author"),
-#'  driver_name = "",
-#'  host = "mysql-rfam-public.ebi.ac.uk",
-#'  database_name = "Rfam",
-#'  user = "rfamro",
-#'  password = "",
-#'  port = 4497
+#'   source        = "select author_id, name, last_name from author",
+#'   dbms          = "MySQL",
+#'   tables        = c("family_author", "author"),
+#'   driver_name   = "",
+#'   host          = "mysql-rfam-public.ebi.ac.uk",
+#'   database_name = "Rfam",
+#'   user          = "rfamro",
+#'   password      = "",
+#'   port          = 4497
 #' )
+#' }
 #'
 fetch_data_from_query <- function(source, dbms, tables,
-                                 driver_name, host, database_name,
-                                 user, password, port) {
-  checkmate::assert_vector(source,
-                           any.missing = FALSE, min.len = 1,
-                           null.ok = FALSE, unique = TRUE
-  )
-  checkmate::assert_character(dbms, any.missing = FALSE, len = 1,
-                              null.ok = FALSE)
+                                  driver_name, host, database_name,
+                                  user, password, port) {
   checkmate::assert_vector(tables,
-                           any.missing = FALSE, min.len = 1,
+                           any.missing = FALSE, min.len = 1L,
                            null.ok = FALSE, unique = TRUE)
-  checkmate::assert_character(driver_name, len = 1, null.ok = FALSE,
-                              any.missing = FALSE)
-  checkmate::assert_character(host, any.missing = FALSE, len = 1,
-                              null.ok = FALSE)
-  checkmate::assert_character(database_name, any.missing = FALSE, len = 1,
-                              null.ok = FALSE)
-  checkmate::assert_character(user, any.missing = FALSE, len = 1,
-                              null.ok = FALSE)
-  checkmate::assert_character(password, any.missing = FALSE, len = 1,
-                              null.ok = FALSE)
-  checkmate::assert_number(port, lower = 1)
 
-  pool <- connect_to_server(dbms, driver_name, host, database_name,
-                           user, password, port)
+  pool <- connect_to_server(
+    dbms, driver_name, host, database_name,
+    user, password, port
+  )
   result <- list()
-  for (query in source) {
+  for (query in source) { # nolint
     table <- identify_table_name(query, tables)
     stopifnot("Could not detect table name from the query" = !is.null(table))
-    result[[table]] <- DBI::dbGetQuery(pool, source)
+    result[[table]] <- DBI::dbGetQuery(pool, source) # nolint
```

Comment:
Why `# nolint`?

Reply:
this linter is related to the `source` variable name. 
I will either use the specific linter name or replace `source` with `src`